internal/btf/btf.go

```diff
@@ -718,7 +718,7 @@ func marshalBTF(types interface{}, strings []byte, bo binary.ByteOrder) []byte {
 	return buf.Bytes()
 }
 
-var haveBTF = internal.FeatureTest("BTF", "5.1", func() (bool, error) {
+var haveBTF = internal.FeatureTest("BTF", "5.1", func() error {
```

Comment:
Now that these no longer return booleans, the `have___` prefix no longer feels intuitive. Having trouble coming up with an alternative convention, though. Don't immediately remember a similar implementation to draw inspiration from. 

Perhaps `feature___` would work better? `featureBTF`, `featureFuncLinkage`, `featureProgAttach`, etc. That doesn't really convey `nil` as being a 'positive' return value either.

Perhaps `test___`? `testBTF`, `testFuncLinkage`, `testProgAttach`. `if testBTF() == nil { }` looks good, but the term `test` already carries other meaning(s).

Reply:
Maybe `check`? I like `feature...` personally.

Reply:
I've tried `feature...` and have decided against it for two reasons. First, it doesn't read clearer than the `have` version unfortunately:
```
	if featureObjName() == nil {
...
	}
```

Second, it means renaming a bunch of tests in addition to the variables, at which point it seems like it's not worth it.

Reply:
I just re-read your comment and realised that you pointed out exactly this problem, oh well...